### PR TITLE
Migrate filename hanlding to own class

### PR DESCRIPTION
**Pull request by vulder**

### CODE REVIEW

File: varats/varats/data/databases/blame_diff_metrics_database.py [hunk 1]
```diff
@@ -111,8 +111,7 @@ def build_report_files_tuple(
     }
 
     failed_report_files: tp.Dict[str, Path] = {
-        MetaReport.get_commit_hash_from_result_file(report.name): report
-        for report in get_failed_revisions_files(
+        ReportFilename(report): report for report in get_failed_revisions_files(
```
Comment (vulder) [inline varats/varats/data/databases/blame_diff_metrics_database.py]:
```suggestion
        ReportFilename(report).commit_hash: report for report in get_failed_revisions_files(
```

File: varats-core/varats/report/report.py [hunk 2]
```diff
@@ -279,99 +197,219 @@ def result_file_has_status(
...
+        "{shorthand}-" + "SUPPL-" + "{project_name}-" + "{binary_name}-" +
+        "{project_version}_" + "{project_uuid}_" + "{info_type}" + "{file_ext}"
+    )
+
+    def __init__(
+        cls: tp.Any, name: str, bases: tp.Tuple[tp.Any], attrs: tp.Dict[str,
+                                                                        tp.Any]
+    ) -> None:
+        super(MetaReport, cls).__init__(name, bases, attrs)
+        #MetaReport.__check_accessor_methods(cls)
```
Comment (boehmseb) [inline varats-core/varats/report/report.py]:
```suggestion
```

File: varats-core/varats/report/report.py [hunk 3]
```diff
@@ -279,99 +197,219 @@ def result_file_has_status(
...
+
+    def __check_required_vars(
+        cls: tp.Any, class_name: str, req_vars: tp.List[str]
+    ) -> None:
+        for var in req_vars:
+            if not hasattr(cls, var):
+                raise NameError((
+                    f"{class_name} does not define "
+                    f"a static variable {var}."
+                ))
```
Comment (boehmseb) [inline varats-core/varats/report/report.py:357]: This could be useful elsewhere in the future.
Should we put it into a utils module?

**Review by boehmseb — APPROVED** [review]

File: varats-core/varats/report/report.py (hunk 3 above)
Comment (vulder) [reply varats-core/varats/report/report.py:357]: You mean the static vars decorator

**Pull request merged**